modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java

```diff
@@ -62,26 +62,30 @@ public static void handleInbound(CarbonMessage cMsg, CarbonCallback callback) {
 
         String protocol = (String) cMsg.getProperty(org.wso2.carbon.messaging.Constants.PROTOCOL);
         if (protocol == null) {
+            cMsg.setProperty(org.wso2.carbon.transport.http.netty.common.Constants.HTTP_STATUS_CODE, 500);
```

Comment:
Please find a alternative solution to set STATUS_CODE as this logic to applicable for Server connectors. Similar to #2458